localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_session.py

```diff
@@ -11,61 +14,135 @@
+            with open(self._configuration_file_path, "r") as df:
                 yaml_configuration_string = df.read()
         except FileNotFoundError:
-            LOG.error("Error: The file lambda debug config " "file '%s' was not found.", file_path)
+            LOG.error(
+                "Error: The file lambda debug config " "file '%s' was not found.",
+                self._configuration_file_path,
+            )
         except IsADirectoryError:
             LOG.error(
                 "Error: Expected a lambda debug config file " "but found a directory at '%s'.",
-                file_path,
+                self._configuration_file_path,
             )
         except PermissionError:
             LOG.error(
                 "Error: Permission denied while trying to read "
                 "the lambda debug config file '%s'.",
-                file_path,
+                self._configuration_file_path,
             )
         except Exception as ex:
             LOG.error(
                 "Error: An unexpected error occurred while reading "
                 "lambda debug config '%s': '%s'",
-                file_path,
+                self._configuration_file_path,
                 ex,
             )
         if not yaml_configuration_string:
             return None
 
-        config = load_lambda_debug_mode_config(yaml_configuration_string)
-        return config
+        self._config = load_lambda_debug_mode_config(yaml_configuration_string)
+        if self._config is not None:
+            LOG.info("Lambda Debug Mode is now enforcing the latest configuration.")
+        else:
+            LOG.warning(
+                "Lambda Debug Mode could not load the latest configuration due to an error, "
+                "check logs for more details."
+            )
+
+    def _config_file_epoch_last_modified_or_now(self) -> int:
+        try:
+            modified_time = os.path.getmtime(self._configuration_file_path)
+            return int(modified_time)
+        except Exception as e:
+            print("Lambda Debug Mode could not access the configuration file: %s", e)
+            epoch_now = int(time.time())
+            return epoch_now
+
+    def _watch_logic(self) -> None:
+        # TODO: consider relying on system calls (watchdog lib for cross-platform support)
+        #  instead of monitoring last modified dates.
+        # Run the first load and signal as initialised.
+        epoch_last_loaded: int = self._config_file_epoch_last_modified_or_now()
+        self._load_lambda_debug_mode_config()
+        self._initialised_event.set()
+
+        # Monitor for file changes in an endless loop: this logic should be started as a daemon.
+        while True:
+            time.sleep(1)
+            epoch_last_modified = self._config_file_epoch_last_modified_or_now()
+            if epoch_last_modified > epoch_last_loaded:
+                epoch_last_loaded = epoch_last_modified
+                self._load_lambda_debug_mode_config()
```

Comment:
I am not a fan of this endless loops, even though the thread is a daemon thread. I would prefer an orderly shutdown of the thread, by using an stopping event as loop condition, and shutting it down on shutdown of the lambda service.

Reply:
changed to `SERVICE_PLUGINS.is_running("lambda")`